packages/web-components/fast-foundation/src/date-picker/date-picker.ts

```diff
@@ -0,0 +1,1262 @@
+     */
+    @observable
+    public monthView: number = new Date().getFullYear();
+
+    /**
+     * The year for the month selector
+     * @public
+     */
+    @observable
+    public yearView: number = Math.floor(new Date().getFullYear() / 10) * 10;
+
+    /**
+     * Dates that are not selectable
+     * @alpha
+     */
+    @attr({ attribute: "disabled-dates" })
+    public disabledDates: string;
+
+    /**
+     * The text field used for date entry
+     * @public
+     */
+    @observable
+    public textField: TextField;
+
+    /**
+     * @internal
+     */
+    @observable
+    public defaultSlottedNodes: Node[];
+
+    /**
+     * Adds the value to the textfield when it's attached to the dom
+     * @param previous - previous textfield element
+     * @param next - updated textfield
+     * @public
+     */
+    public textFieldChanged(previous: TextField, next: TextField) {
+        // Once the textfield is loaded, format its value
+        if (!previous && !!next && this.value) {
+            this.textField.value = this.value;
+        }
+    }
+
+    /**
+     * The hour select listbox
+     * @public
+     */
+    public hourSelect: ListboxElement;
+
+    /**
+     * The minute select listbox
+     * @public
+     */
+    public minuteSelect: ListboxElement;
+
+    /**
+     * The meridian select listbox
+     * @public
+     */
+    public meridianSelect: ListboxElement;
+
+    /**
+     * The Calendar month
+     * @public
+     */
+    @observable
+    public calendarMonth: number = new Date().getMonth() + 1;
+
+    /**
+     * Handles changing the calendar view month
+     * @param previous - the current month value
+     * @param next - the next month value
+     * @public
+     */
+    public calendarMonthChanged(previous: number, next: number) {
+        this.setCalendarTitle();
+    }
+
+    /**
+     * The Calendar year
+     * @public
+     */
+    @observable
+    public calendarYear: number = new Date().getFullYear();
+
+    /**
+     * Handles changing the calendar view year
+     * @param previous - the current calendar year view
+     * @param next - the next calendar year view
+     * @public
+     */
+    public calendarYearChanged(previous: number, next: number) {
+        this.setCalendarTitle();
+    }
+
+    /**
+     * Sets the calendar title text
+     * @public
+     */
+    public setCalendarTitle() {
+        if (this.calendarMonth && this.calendarYear) {
+            this.calendarTitle = this.dateFormatter.getDate(
+                `${this.calendarMonth}/1/${this.calendarYear}`,
+                { month: "long", year: "numeric" }
+            );
+        }
+    }
+
+    /**
+     * The title text for the calendar
+     */
+    @observable
+    public calendarTitle: string = "";
+
+    /**
+     * Keyboard handler for the calendar title
+     * @param e - Keyboard event fired
+     * @returns should bubble
+     * @public
+     */
+    public handleCalendarTitleKeydown(e: KeyboardEvent): void {
```

Comment:
I think this is marking all keystrokes with prevent default, probably what is blocking tabbing, return true if it makes it past your switch statement I think.